cypress/support/MonitorDocumentChanges.js

```diff
@@ -34,6 +35,16 @@ import * as helpers from './Helpers.js'
 
   function MillisecondsSinceLastChange() {
     if (new Date().getTime() > lastMonitorTime + 50) LookForChange()
+    
+    // Some operations are known to take a long time without rendering, so while this count
+    // is greater than zero, we will reset our lastChangeTime and return zero which will
+    // effectively cause a wait until the specified number of changes pass and then resume
+    // our normal monitoring for stable DOM.
+    if (waitTillNChangesOccur > 0) { 
+      lastChangeTime = new Date().getTime()
```

Comment:
What are the cases this is needed?  Would rather use the builtin Cypress `timeout` instead as this is what it's for.

Reply:
[Bug 1950](https://dialearn.visualstudio.com/BLIS/_workitems/edit/1950): Save Train Dialog takes too long to render grid results without spinner nor discernable API call activity

So this is fudging for what I believe is a product bug. Which is why I filed the bug. 

Saving the Train Dialog has consistently had odd rendering issues from the time I started testing it. They get fixed, only for a new one to crop up a few weeks later with new changes.

The reason I like this solution is that it is definitive and precise. We can see how many changes constitute a render in the logs, and just wait for them to occur before we resume normal operations. Also it is implemented at a single step of a test, saving the Train Dialog, as opposed to the other multiple possible points of failure.

The biggest reason is that it stays consistent of the philosophy and patterns behind the DOM Monitoring code, and that is to block test progress until we know the DOM is stable after whatever recent changes were triggered by UI activity. Last I heard you were objecting to this DOM Monitoring code so this too will need to be discussed when we have our meeting on Wednesday.